src/crimson/mgr/client.h

```diff
@@ -30,7 +30,8 @@ class WithStats {
   virtual ~WithStats() {}
 };
 
-class Client : public crimson::net::Dispatcher {
+class Client : public crimson::net::Dispatcher,
+	       public crimson::common::Gated {
```

Comment:
why shall we inherit from `Gated`? what was your motivation? have your tried to apply [LSP](https://en.wikipedia.org/wiki/Liskov_substitution_principle) to it? can we make it a private member variable instead?